R/slice.R

```diff
@@ -165,6 +174,93 @@ vec_as_index <- function(i, n, names = NULL) {
   vec_assert(n, integer(), 1L)
   .Call(vctrs_as_index, i, n, names)
 }
+#' @rdname vec_as_index
+#' @export
+vec_as_position <- function(i, n, names = NULL) {
+  if (is.object(i)) {
+    if (vec_is_coercible(i, int())) {
+      i <- vec_cast(i, int())
+    } else if (vec_is_coercible(i, chr())) {
+      i <- vec_cast(i, chr())
+    }
```

Comment:
If an object isn't coercible to integer or character, I think we should error here, rather than using the underlying base type.

Reply:
oops, agreed!